File: CPP/Algorithms/SortSearch/DFSadjMatrix.cpp
// C++ implementation of the approach
#include <bits/stdc++.h>
using namespace std;

// adjacency matrix
vector<vector<int> > adj;


void addEdge(int x, int y)
{
	adj[x][y] = 1;
	adj[y][x] = 1;
}


void dfs(int start, vector<bool>& visited)
{
	cout << start << " ";
	visited[start] = true;

	for (int i = 0; i < adj[start].size(); i++) {

		if (adj[start][i] == 1 && (!visited[i])) {
			dfs(i, visited);
		}
	}
}

int main()
{

	int v = 5;
	//int e = 4;

	adj = vector<vector<int> >(v, vector<int>(v, 0));

	addEdge(0, 1);
	addEdge(0, 2);
	addEdge(0, 3);
	addEdge(2, 3);
	addEdge(2, 4);


	vector<bool> visited(v, false);
	dfs(0, visited);
  cout<<endl;
}

File: JavaScript/Algorithms/SortSearch/DFSadjMatrix.js
let ans=""; 
class Graph {
constructor(v, e) {
	this.v = v; // number of vertices
	this.e = e; // number of edges

	// initialize the adjacency matrix with 0s
	this.adj = Array.from(Array(v), () => new Array(v).fill(0));
}

// function to add an edge to the graph
addEdge(start, end) {
	// considering a bidirectional edge
	this.adj[start][end] = 1;
	this.adj[end][start] = 1;
}

// function to perform DFS on the graph
DFS(start, visited) {

	ans = ans +start + " ";
	visited[start] = true;

	for (let i = 0; i < this.v; i++) {

	if (this.adj[start][i] === 1 && !visited[i]) {
		this.DFS(i, visited);
	}
	}
}
}


const v = 5;
const e = 4; 

// create the graph
const G = new Graph(v, e);
G.addEdge(0, 1);
G.addEdge(0, 2);
G.addEdge(0, 3);
G.addEdge(0, 4);


const visited = new Array(v).fill(false);

G.DFS(0, visited); 
console.log(ans);
